interbtc-spec/docs/source/spec/refund.rst

```diff
@@ -30,3 +30,51 @@ Security
 --------
 
 - Unique identification of Bitcoin payments: :ref:`op-return`
+
+Functions
+~~~~~~~~~
+
+.. _executeRefund:
+
+executeRefund
+--------------
+
+This function finalizes a refund, also referred to as a user failsafe. 
+It is typically called by the vault client that performed the refund.
+
+Specification
+.............
+
+*Function Signature*
+
+``execute_refund(caller, refundId, merkleProof, rawTx)``
+
+*Parameters*
+
+* ``caller``: address of the user finalizing the refund. Typically the vault client that performed the refund.
+* ``refundId``: the unique hash created during the internal ``requestRefund`` function.
+* ``merkleProof``: merkle tree path (concatenated LE SHA256 hashes).
+* ``rawTx``: raw Bitcoin transaction of the refund payment, including the transaction inputs and outputs.
+
+*Events*
+
+* ``ExecuteRefund(refundId, issuer, vault, amount, fee)``
+
+*Preconditions*
+
+* The function call MUST be signed be *someone*, i.e. not necessarily the vault client that performed the refund.
+* The BTC Parachain status in the :ref:`security` component MUST NOT be set to ``SHUTDOWN:2``.
+* A *pending* ``RefundRequest`` MUST exist with an id equal to ``refundId``.
+* ``refundRequest.completed`` MUST be ``false``.
+* The ``rawTx`` MUST decode to a valid transaction that transfers the amount specified in the ``RefundRequest`` struct. It MUST be a transaction to the correct address, and provide the expected OP_RETURN, based on the ``RefundRequest``.
+* The ``merkleProof`` MUST contain a valid proof of of ``rawTX``.
+* The bitcoin payment MUST have been submitted to the relay chain, and MUST have sufficient confirmations.
+* The ``vault.status`` MUST be ``active``.
+* The ``vault.isBanned()`` MUST return ``false``.
+* The refunding vault MUST have enough collateral to mint an amount equal to the refund fee.
+
+*Postconditions*
+
+* The ``vault.issuedTokens`` MUST increase by ``fee``.
+* The ``TotalSupply`` in the :ref:`treasury-module` MUST increase by ``fee``.
+* The vault's ``SLA`` MUST increase by the :ref:`sla` score of ``Refund``.
```

Comment:
* The vault spendable interBTC balance MUST increase by ``fee``
* ``refundRequest.completed`` MUST be set to ``true``.